fluent/src/bundle.js

```diff
@@ -207,35 +207,38 @@ export default class FluentBundle {
    *         // [<ReferenceError: Unknown variable: name>]
    *     }
    *
+   * If `errors` is omitted, the first encountered error will be thrown.
+   *
    * @param   {string|Array} pattern
    * @param   {?Object} args
    * @param   {?Array} errors
    * @returns {string}
    */
   formatPattern(pattern, args, errors) {
-    // Resolve a simple pattern without creating a scope.
+    // Resolve a simple pattern without creating a scope. No error handling is
+    // required; by definition simple patterns don't have placeables.
     if (typeof pattern === "string") {
       return this._transform(pattern);
     }
 
     // Resolve a complex pattern.
-    if (Array.isArray(pattern)) {
-      let scope = this._createScope(args, errors);
-      try {
-        let value = resolveComplexPattern(scope, pattern);
-        return value.toString(scope);
-      } catch (err) {
+    let scope = this._createScope(args, errors);
+    try {
+      let value = resolveComplexPattern(scope, pattern);
+      return value.toString(scope);
+    } catch (err) {
+      if (scope.errors) {
```

Comment:
Here, `scope.error` controls whether _fatal_ errors should be thrown or should produce `{???}`.